File: var/www/cgi-bin/testGET.php
#!/usr/bin/php
<?php
// testGET.php

$response_body = "<html><head><title>My CGI Script</title></head><body><h1>GET REQUEST</h1></body></html>";

// Add the environment variables to the response body
$response_body .= "<h2>Environment Variables of the cgi get script </h2>";
$response_body .= "<ul>";
foreach (array_keys(getenv()) as $key) {
    $value = getenv($key);
    $response_body .= "<li><strong>" . htmlspecialchars($key) . ":</strong> " . htmlspecialchars($value) . "</li>";
}
$response_body .= "</ul>";

echo "Status: 200 OK\r\n";
echo "Content-Type: text/html\r\n";
echo "Content-Length: " . strlen($response_body) . "\r\n\r\n";
echo $response_body;
?>

File: var/www/cgi-bin/testGET.pl
#!/usr/bin/perl

use strict;
use warnings;
# use CGI;

sub handle_get {
    # my $cgi = CGI->new;

    # Prepare the response body
    my $response_body = "<html><head><title>GET Request</title></head><body>";
    $response_body .= "<h1>Received GET data</h1>";
    $response_body .= "<ul>";

    # Iterate over all environment variables
    foreach my $key (sort keys %ENV) {
        my $value = $ENV{$key};
        $response_body .= "<li><strong>$key</strong>: $value</li>";
    }

    $response_body .= "</ul>";
    $response_body .= "</body></html>";

    # Calculate the content length of the response body
    my $response_content_length = length($response_body);

    # Set up headers for the response
    print "Status: 200 OK\r\n";
    print "Content-Type: text/html\r\n";
    print "Content-Length: $response_content_length\r\n";
    print "\r\n";

    # Output the response body
    print $response_body;
}

sub main {
    # Check if the request method is GET
    my $method = $ENV{'REQUEST_METHOD'} || 'GET';
    if ($method eq 'GET') {
        handle_get();
    } else {
        # Set up headers for the response
        my $error_message = "<html><head><title>405 Method Not Allowed</title></head><body><h1>405 Method Not Allowed</h1></body></html>";
        print "Status: 405 Method Not Allowed\r\n";
        print "Content-Type: text/html\r\n";
        print "Content-Length: " . length($error_message) . "\r\n";
        print "\r\n";
        # Output the error message
        print $error_message;
    }
}

main();
